### PR TITLE
introduce Sponge instructions `absorb_init`, `absorb`, and `squeeze`

### LINKED ISSUE #166: add `sponge_` instructions
Hashing input of variable length or producing many pseudo-random field elements is currently more difficult than it needs to be. Critically, it is next to impossible to mimic the rust code's hashing behavior used for verifying proofs. This effectively prevents efficient recursive verification.

Add instructions to use the permutation used for all things hashing as a [Sponge](https://keccak.team/sponge_duplex.html). Those are probably:

- `sponge_init`
- `sponge_absorb`
- `sponge_squeeze`

Supersedes #157.

### PR DESCRIPTION
**Pull request by jan-ferdinand**
Instruction `absorb_init` both initializes the Sponge state and absorbs the first 10 elements. It is an error to execute `absorb` or `squeeze` before the first execution of `absorb_init`.

Only at most one Sponge state exists at any given time. It is possible to re-initialize the Sponge's state by executing `absorb_init`.

The Sponge instructions are independent of instruction `hash`.

For arithmetization reasons, two new base columns and two new extension columns are added, one of each in the Processor Table and the Hash Table respectively.

Also:
- use twenty-first v0.14

Fix #166

### CODE REVIEW

**Review by aszepieniec — APPROVED** [review]

File: specification/src/processor-table.md [hunk 1]
```diff
@@ -175,8 +179,9 @@ The following constraints apply to every pair of rows.
 1. The running product for the OpStack Table absorbs the next row with respect to challenges 🍋, 🍊, 🍉, and 🫒 and indeterminate 🪤.
 1. The running product for the RAM Table absorbs the next row with respect to challenges 🍍, 🍈, 🍎, and 🌽 and indeterminate 🛋.
 1. The running product for the JumpStack Table absorbs the next row with respect to challenges 🍇, 🍅, 🍌, 🍏, and 🍐 and indeterminate 🧴.
-1. If the current instruction in the next row is `hash`, the running evaluation “to Hash Table” absorbs the next row with respect to challenges 🧄0 through 🧄9 and indeterminate 🪣. Otherwise, it remains unchanged.
-1. If the current instruction is `hash`, the running evaluation “from Hash Table” absorbs the next row with respect to challenges 🫑0 through 🫑4 and indeterminate 🪟. Otherwise, it remains unchanged.
+1. If the current instruction in the next row is `hash`, the running evaluation “Hash Input absorbs the next row with respect to challenges 🧄0 through 🧄9 and indeterminate 🚪. Otherwise, it remains unchanged.
```
[the remark this answers is not in the document]
Comment (jan-ferdinand) [reply specification/src/processor-table.md]: Thanks for being so thorough. :+1:

File: triton-opcodes/Cargo.toml [hunk 2]
```diff
@@ -23,7 +23,7 @@ version = "1"
 default-features = false
 
 [dependencies]
-twenty-first = "0.11"
+twenty-first = "0.14"
```
Comment (sshine) [inline triton-opcodes/Cargo.toml:26]:
```suggestion
twenty-first = "0.14.1"
```

File: constraint-evaluation-generator/Cargo.toml [hunk 3]
```diff
@@ -23,6 +23,6 @@ version = "1"
 default-features = false
 
 [dependencies]
-twenty-first = "0.11"
+twenty-first = "0.14"
```
Comment (sshine) [inline constraint-evaluation-generator/Cargo.toml:26]:
```suggestion
twenty-first = "0.14.1"
```

File: triton-profiler/Cargo.toml [hunk 4]
```diff
@@ -15,7 +15,7 @@ keywords = ["triton-vm", "stark", "profiler"]
 categories = ["cryptography", "mathematics"]
 
 [dependencies]
-twenty-first = "0.11"
+twenty-first = "0.14"
```
Comment (sshine) [inline triton-profiler/Cargo.toml:18]:
```suggestion
twenty-first = "0.14.1"
```

File: triton-vm/Cargo.toml [hunk 5]
```diff
@@ -30,8 +30,8 @@ version = "1"
 default-features = false
 
 [dependencies]
-triton-opcodes = "0.13"
-twenty-first = "0.11"
+triton-opcodes = { path = "../triton-opcodes" }
+twenty-first = "0.14"
```
Comment (sshine) [inline triton-vm/Cargo.toml:34]:
```suggestion
twenty-first = "0.14.1"
```

**Review by sshine — APPROVED** [review]

File: triton-vm/Cargo.toml (hunk 5 above)
Comment (jan-ferdinand) [reply triton-vm/Cargo.toml:34]: I'll update the dependencies with the release of the next version Triton VM, after the merge. :+1:

**Pull request merged**